gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java

```diff
@@ -1233,12 +1251,200 @@ protected TypeNode getTransportOperationsStubType(Service service) {
     return transportOpeationsStubType;
   }
 
+  protected static LambdaExpr createRequestMutatorClassInstance(
+      Method method, ImmutableMap<String, Message> messageTypes) {
+    List<Statement> bodyStatements = new ArrayList<>();
+    VariableExpr requestVarExpr = createRequestVarExpr(method);
+
+    Reference requestBuilderRef =
+        VaporReference.builder()
+            .setEnclosingClassNames(method.inputType().reference().name())
+            .setName("Builder")
+            .setPakkage(method.inputType().reference().pakkage())
+            .build();
+
+    TypeNode requestBuilderType = TypeNode.withReference(requestBuilderRef);
+
+    VariableExpr requestBuilderVarExpr =
+        VariableExpr.builder()
+            .setVariable(
+                Variable.builder().setName("requestBuilder").setType(requestBuilderType).build())
+            .setIsDecl(true)
+            .build();
+
+    MethodInvocationExpr setRequestBuilderInvocationExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(requestVarExpr)
+            .setMethodName("toBuilder")
+            .setReturnType(requestBuilderType)
+            .build();
+
+    Expr requestBuilderExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(requestBuilderVarExpr)
+            .setValueExpr(setRequestBuilderInvocationExpr)
+            .build();
+
+    bodyStatements.add(ExprStatement.withExpr(requestBuilderExpr));
+
+    VariableExpr returnBuilderVarExpr =
+        VariableExpr.builder()
+            .setVariable(
+                Variable.builder().setName("requestBuilder").setType(requestBuilderType).build())
+            .setIsDecl(false)
+            .build();
+
+    MethodInvocationExpr.Builder returnExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(returnBuilderVarExpr)
+            .setMethodName("build");
+
+    createRequestMutatorBody(method, messageTypes, bodyStatements, returnBuilderVarExpr);
+
+    return LambdaExpr.builder()
+        .setArguments(requestVarExpr.toBuilder().setIsDecl(true).build())
+        .setBody(bodyStatements)
+        .setReturnExpr(returnExpr.build())
+        .build();
+  }
+
+  @VisibleForTesting
+  static List<Statement> createRequestMutatorBody(
+      Method method,
+      ImmutableMap<String, Message> messageTypes,
+      List<Statement> bodyStatements,
+      VariableExpr returnBuilderVarExpr) {
+
+    if (!shouldGenerateRequestMutator(method, messageTypes)) {
+      return bodyStatements;
+    }
+    Message methodRequestMessage = messageTypes.get(method.inputType().reference().fullName());
+    method.autoPopulatedFields().stream()
```

Comment:
Might be missing a bit of context, but I'm a bit lost on this part. Retrieve the autoPopulatedFields from the RPC. I'm guessing there are more autopopulated fields than just `request_id`? 

Or is b/c some request_id fields don't full match all the conditions: 1. field_info = UUID, 2. optional, and 3. string?

Reply:
I see, I see some changes were done in an earlier PR regarding parsing the service yaml file. So just to clarify my understanding, `method.autoPopulatedfields()` is just a list of fields from the service yaml file.